website/static/js/registrationEditorExtensions.js

```diff
@@ -131,19 +148,110 @@ ko.bindingHandlers.osfUploader = {
     init: osfUploader
 };
 
-var Uploader = function(data) {
+var Uploader = function(question) {
 
     var self = this;
-    self._orig = data;
 
-    self.selectedFileName = ko.observable('no file selected');
+    question.showUploader = ko.observable(false);
+    self.selectedFile = ko.observable(null);
+    self.selectedFile.subscribe(function(file) {
+        if (file) {
+            question.extra({
+                selectedFileName: file.data.name,
+                viewUrl: '/project/' + file.data.nodeId + '/files/osfstorage' + file.data.path,
+                hasSelectedFile: true
+            });
+        }
+        else {
+            question.extra({
+                selectedFileName: 'no file selected'
+            });
+            question.value('no file selected');
+        }
+    });
+    self.hasSelectedFile = ko.computed(function() {
+        return !!(question.extra().viewUrl);
+    });
+    self.unselectFile = self.selectedFile.bind(null, null);
+
     self.filePicker = null;
 
+    self.preview = function() {
+        var value = question.value();
+        if (!value || value === 'no file selected') {
+            return 'no file selected';
+        }
+        else {
+            var extra = question.extra();
+            return $('<a target="_blank" href="' + extra.viewUrl + '">' + extra.selectedFileName + '</a>');
```

Comment:
Avoid writing HTML (with user-entered values) in KO. If doing so is the only way to avoid being pecked to death by parakeets, `$osf.htmlEscape` is a minimum precaution for `filename`, etc.

(For the URL it gets a bit trickier; you don't want to HTML escape ampersand url parameters, for example)